backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/acct/api/impl/ElementValueDAO.java

```diff
@@ -45,4 +51,76 @@ public I_C_ElementValue retrieveById(@CacheCtx final Properties ctx, final int e
 		}
 		return InterfaceWrapperHelper.create(ctx, elementValueId, I_C_ElementValue.class, ITrx.TRXNAME_None);
 	}
+
+	@Override
+	public @NonNull ImmutableSet<ElementValueId> getElementValueIdsBetween(@NonNull final String valueFrom, @NonNull final String valueTo)
+	{
+		/*
+Implementation detail: Similar to: org.compiere.apps.search.Info.Worker.loadData, which is called from org.compiere.acct.AcctViewer.actionButton, when searching from an account.
+
+ValueFrom sql:
+SELECT ev.c_elementvalue_id, ev.value, ev.name
+FROM c_elementvalue ev
+WHERE TRUE
+  AND IsActive = 'Y'
+  AND (IsSummary = 'N' OR IsSummary IS NULL)
+  AND C_Element_ID = 1000000
+  AND Value LIKE '10%'
+ORDER BY value Asc
+LIMIT 1
+;
+
+
+ValueTo sql:
+SELECT ev.c_elementvalue_id, ev.value, ev.name
+FROM c_elementvalue ev
+WHERE TRUE
+  AND IsActive = 'Y'
+  AND (IsSummary = 'N' OR IsSummary IS NULL)
+  AND C_Element_ID = 1000000
+  AND Value LIKE '11%'
+ORDER BY value DESC
+LIMIT 1
+;
+
+
+Query similar to org.compiere.acct.AcctViewerData.appendAccountWhereClause:
+SELECT ce.c_elementvalue_id
+FROM c_elementvalue ce
+WHERE (
+          LPAD(trim(ce.Value), 20, '0')
+              BETWEEN
+             	 (SELECT LPAD(trim(ev.Value), 20, '0') FROM C_ElementValue ev WHERE ev.C_ElementValue_ID = $idFrom)
+              AND
+             	 (SELECT LPAD(trim(ev.Value), 20, '0') FROM C_ElementValue ev WHERE ev.C_ElementValue_ID = $idTo)
+          )
+;
+
+Observation, even if the original query uses LPAD, RPAD gives better results from my testing
```

Comment:
IMHO the whole comment intoduces more noise and FUD than helps. pls remove it

Reply:
done

Reply:
removed comment